plugins/modules/win_net_adapter_feature.ps1

```diff
@@ -0,0 +1,127 @@
+#!powershell
+#
+# (c) 2020, ライトウェルの人 <@jirolin>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible. If not, see <http://www.gnu.org/licenses/>.
+#
+# WANT_JSON
+# POWERSHELL_COMMON
+# temporarily disable strictmode, for this module only
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+
+$Interface_name = [string[]]
+$componentID_name = [string[]]
+$params = Parse-Args $args -supports_check_mode $true
+#$Interface_name = Get-AnsibleParam -obj $params -name "Interface" -type "str"
+$Interface_name = Get-AnsibleParam -obj $params -name "Interface"
+$state = Get-AnsibleParam -obj $params -name state -type "str"
+#$componentID_name = Get-AnsibleParam -obj $params -name "componentID" -type "str"
+$componentID_name = Get-AnsibleParam -obj $params -name "componentID"
+$check_mode = Get-AnsibleParam -obj $params -name "_ansible_check_mode" -type "bool" -default $false
+
+$result = @{
+    changed = $false
+}
+
+If(!$Interface_name) { Fail-Json -message "Absolute Interface is not specified for Interface" }
+If(!$state) { Fail-Json -message "Absolute state is not specified for state" }
+If(!$componentID_name) { Fail-Json -message "Absolute componentID is not specified for componentID" }
+
+If ($state -eq 'enable'){
+    $state = $true
+}ElseIf ($state -eq 'disable') {
+    $state = $false
+}Else {
+    Fail-Json -message "Specify the state as 'enable' or 'disable'"
+}
+
+If($Interface_name -is [string]) {
+    If($Interface_name.Length -gt 0) {
+        $Interfaces = @($Interface_name)
+    }
+    Else {
+        $Interfaces = @()
+    }
+}
+Else {
+        $Interfaces = $Interface_name
+}
+
+If($componentID_name -is [string]) {
+    If($componentID_name.Length -gt 0) {
+        $componentIDs = @($componentID_name)
+    }
+    Else {
+        $componentIDs = @()
+    }
+}
+Else {
+    $componentIDs = $componentID_name
+}
```

Comment:
So we want to avoid adding any new modules that use the `Legacy` wrapper. There are quite a few missing features when you use this util compared to the replacement. Also when you use the replacement you can get rid of a lot of boilerplate checks that are currently present in the module like so.

Keep in mind there are a few things you need to do differently compared to `Ansible.ModuleUtils.Legacy` like:

* Any calls to `Fail-Json` need to be replaced with `$module.FailJson("msg")`
* Any calls to `Exit-Json` need to be replaced with `$module.ExitJson()`
* Setting a `$result.changed = $true` is now `$module.Result.changed = $true`

```suggestion
$spec = @{
    options = @{
        interface = @{ type = 'list'; elements = 'str'; required = $true }
        state = @{ type = 'str'; choices = 'disabled', 'enabled'; default = 'enabled' }
        component_id = @{ type = 'list'; elements = 'str'; required = $true }
    }
    supports_check_mode = $true
}
$module = [Ansible.Basic.AnsibleModule]::Create($args, $spec)

$interface = $module.Params.interface
$state = $module.Params.state
$component_id = $module.Params.component_id
$check_mode = $module.CheckMode
```

Reply:
Thank you!
I need to check the docs carefully and to execute tests again.

Reply:
No worries, it unfortunately isn't a simple swap over but we tried to make it as painless as we could while adding support for newer features.